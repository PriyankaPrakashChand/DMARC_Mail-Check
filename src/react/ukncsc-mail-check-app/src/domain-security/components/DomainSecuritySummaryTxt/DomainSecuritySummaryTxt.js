import React, { Component } from 'react';
import { Grid } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import { MailCheckMessage } from 'common/components';
import {
  DomainSecurityRecord,
  DomainSecuritySummaryMessages,
  DomainSecurityTitle,
} from 'domain-security/components';
import { DomainSecurityLocationContext } from 'domain-security/context';

const shouldShowRecords = txt => txt && !!txt.records;

const shouldShowMessages = txt =>
  txt && !txt.pending && !txt.loading && !txt.error;

export default class DomainSecuritySummary extends Component {
  componentDidMount() {
    const {
      domainId,
      fetchDomainSecurityTxt,
      getDomainSecurityTxt,
    } = this.props;

    if (!getDomainSecurityTxt(domainId)) {
      fetchDomainSecurityTxt(domainId);
    }
  }

  render() {
    const { domainId, description, getDomainSecurityTxt, type } = this.props;
    const txt = getDomainSecurityTxt(domainId);

    return (
      <Grid stackable>
        <Grid.Row>
          <Grid.Column width="4">
            <DomainSecurityTitle as="h2" title={type} {...txt} />
          </Grid.Column>
          <Grid.Column width="8">
            <p>{description}</p>

            {shouldShowRecords(txt) &&
              txt.records.map(({ record }, i) => (
                <DomainSecurityRecord
                  type={type}
                  inheritedFrom={txt.inheritedFrom}
                  key={i}
                >
                  {record}
                </DomainSecurityRecord>
              ))}

            {shouldShowMessages(txt) && (
              <React.Fragment>
                <p>
                  <DomainSecurityLocationContext.Consumer>
                    {location => (
                      <Link
                        to={`/${location}/${domainId}/${type.toLowerCase()}`}
                      >
                        View more information
                      </Link>
                    )}
                  </DomainSecurityLocationContext.Consumer>
                </p>
                <DomainSecuritySummaryMessages
                  type={type}
                  failures={txt.failures && txt.failures.length}
                  warnings={txt.warnings && txt.warnings.length}
                  inconclusives={txt.inconclusives && txt.inconclusives.length}
                />
              </React.Fragment>
            )}

            {txt &&
              txt.pending && (
                <MailCheckMessage info>
                  {`We're currently checking your ${type} configuration. This usually takes less than 15 minutes, so check back soon.`}
                </MailCheckMessage>
              )}

            {txt &&
              txt.error && (
                <MailCheckMessage error>
                  There was a problem retrieving the {type} summary for this
                  domain: {txt.error.message}
                </MailCheckMessage>
              )}
          </Grid.Column>
        </Grid.Row>
      </Grid>
    );
  }
}
