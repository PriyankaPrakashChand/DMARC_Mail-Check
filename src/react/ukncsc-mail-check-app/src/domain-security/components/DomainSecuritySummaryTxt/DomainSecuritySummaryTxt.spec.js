import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from 'react-testing-library';
import 'dom-testing-library/extend-expect';

import DomainSecuritySummaryTxt from './DomainSecuritySummaryTxt';

const fetchDomainSecurityTxt = jest.fn();
const getDomainSecurityTxt = jest.fn();

describe('DomainSecuritySummaryTxt', () => {
  let container;

  afterEach(() => jest.resetAllMocks());

  describe('when no records are provided', () => {
    beforeEach(() => {
      ({ container } = render(
        <MemoryRouter>
          <DomainSecuritySummaryTxt
            domainId={123}
            fetchDomainSecurityTxt={fetchDomainSecurityTxt}
            getDomainSecurityTxt={getDomainSecurityTxt}
            type="SPF"
          />
        </MemoryRouter>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should not display a record', () =>
      expect(
        container.getElementsByClassName('DomainSecurityRecord')
      ).toHaveLength(0));
  });
});
