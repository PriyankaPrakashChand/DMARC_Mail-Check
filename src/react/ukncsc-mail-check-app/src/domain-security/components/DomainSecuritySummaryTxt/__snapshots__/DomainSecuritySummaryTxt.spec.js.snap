// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainSecuritySummaryTxt when no records are provided it should match the snapshot 1`] = `
<div>
  <div
    class="ui stackable grid"
  >
    <div
      class="row"
    >
      <div
        class="four wide column"
      >
        <h2
          class="ui header DomainSecurityTitle--header"
        >
          SPF
        </h2>
        <div
          class="DomainSecurityTitle--icon"
        />
      </div>
      <div
        class="eight wide column"
      >
        <p />
      </div>
    </div>
  </div>
</div>
`;
